pkg/reconciler/dependencybuild/buildrecipeyaml.go

```diff
@@ -0,0 +1,481 @@
+        type: string
+      - name: TAG
+        type: string
+      - name: IMAGE
+        type: string
+        description: Build image
+      - name: GOALS
+        description: maven goals to run
+        type: array
+        default:
+          - -DskipTests
+          - clean
+          - install
+          - -Denforcer.skip
+      - name: MAVEN_MIRROR_URL
+        description: The Maven repository mirror url
+        type: string
+        default: http://localhost:2000/maven2
+      - name: SERVER_USER
+        description: The username for the server
+        type: string
+        default: ""
+      - name: SERVER_PASSWORD
+        description: The password for the server
+        type: string
+        default: ""
+      - name: PROXY_USER
+        description: The username for the proxy server
+        type: string
+        default: ""
+      - name: PROXY_PASSWORD
+        description: The password for the proxy server
+        type: string
+        default: ""
+      - name: PROXY_PORT
+        description: Port number for the proxy server
+        type: string
+        default: ""
+      - name: PROXY_HOST
+        description: Proxy server Host
+        type: string
+        default: ""
+      - name: PROXY_NON_PROXY_HOSTS
+        description: Non proxy server host
+        type: string
+        default: ""
+      - name: PROXY_PROTOCOL
+        description: Protocol for the proxy ie http or https
+        type: string
+        default: "http"
+      - name: CONTEXT_DIR
+        type: string
+        description: >-
+          The context directory within the repository for sources on
+          which we want to execute maven goals.
+        default: "."
+      - name: ENFORCE_VERSION
+        type: string
+        description: >-
+          Some builds are incorrectly tagged with a snapshot version, rather
+          than the release version. If this is set then the version will be
+          updated to the release version before the build.
+        default: ""
+      - name: IGNORED_ARTIFACTS
+        type: string
+        description: >-
+          Comma seperated list of artifact names that should not be deployed or checked for contaminants.
+        default: ""
+    steps:
+      - name: git-clone
+        image: "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:v0.21.0"
+        resources:
+          requests:
+            memory: "128Mi"
+            cpu: "10m"
+          limits:
+            memory: "512Mi"
+            cpu: "300m"
+        args:
+          - -path=$(workspaces.source.path)
+          - -url=$(params.URL)
+          - -revision=$(params.TAG)
+      - name: settings
+        image: "registry.access.redhat.com/ubi8/ubi:8.5"
+        securityContext:
+          runAsUser: 0
+        resources:
+          requests:
+            memory: "128Mi"
+            cpu: "10m"
+          limits:
+            memory: "512Mi"
+            cpu: "300m"
+        script: |
+          #!/usr/bin/env bash
```

Comment:
I don't really like this approach long term, but I am fine with it for now (we now have bash embedded in yaml embedded in golang, ugh). Once this is in, the gradle work is in, and we have better builder images I will revisit this. 

There is likely to be a lot of common code between Maven/Gradle, and also things that are not necessary any more that can be removed.

Reply:
Certainly if I'm in the minority wrt not wanting to add/move more yaml files laying around that confuse people, I'll pivot.  But at least for me it is easy to traverse between the consuming code in dependencybuild.go and the variables defined in the new go file in my editor.  And as you noted, as this is suppose to be temporary, this is how I chose to balance the competing concerns, in particular "isolating" future throw away code in golang file we know nobody will mistakingly start referencing in infra-deployments or build-definitions.  I'm also good with pivoting the a golang only approach, though as @Michkov noted that chunk of golang code was getting long and tedious given what is involved with seeding the sidecar.

Certainly better builder images and consolidation of code could help going back to say a golang only approach.

Reply:
I mean, you said you were fine with it ... I guess I'm just trying to say I get not being super enamored with it :-).  At least for me, some of the other choices bothered me more than this one.

Reply:
Yea, there does not seem to be any super good choices here, although given that we now have 4 of these I am hoping that we can share enough code/add utility methods so the code based solution does not end up being too bad. 

This is something to deal with later though.

BTW did you consider using https://pkg.go.dev/embed ? As a Java programmer my first impulse would have been to embed this in the file and load it from the class path.

Reply:
I did but this was at the crux in my "I want to reduce the number of yaml files" motivation.